From: Eric Dumazet <edumazet@google.com>
Date: Thu, 5 Dec 2013 21:44:27 -0800
Subject: gro: small napi_get_frags() optim
Patch-mainline: v3.14-rc1
Git-commit: 84b9cd633bc35a028b313178829ee313525f6892
References: bug#922851 FATE#318528

Remove one useless conditional branch :
napi->skb is NULL, so nothing bad can happen.

Signed-off-by: Eric Dumazet <edumazet@google.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 net/core/dev.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/net/core/dev.c
+++ b/net/core/dev.c
@@ -4001,8 +4001,7 @@ struct sk_buff *napi_get_frags(struct na
 
 	if (!skb) {
 		skb = netdev_alloc_skb_ip_align(napi->dev, GRO_MAX_HEAD);
-		if (skb)
-			napi->skb = skb;
+		napi->skb = skb;
 	}
 	return skb;
 }
